distributed/s3.py

```diff
@@ -0,0 +1,34 @@
+import boto3
+from dask.imperative import Value
+from distributed import default_executor
+from toolz import curry
+
+DEFAULT_PAGE_LENGHT = 1000
+
+
+def get_list_of_summary_objects(bucket_name, prefix='', page_size=DEFAULT_PAGE_LENGHT):
+    s3 = boto3.resource('s3')
+    return [summary_object for summary_object in
+            s3.Bucket(bucket_name).objects.filter(Prefix=prefix).page_size(page_size)]
+
+
+def read_content_from_keys(bucket, key):
+    import boto3
+    s3 = boto3.resource('s3')
+    return s3.Object(bucket, key).get()['Body'].read()
```

Comment:
Do we need to pass around authentication?  What happens if the user only has credentials on their head node but the reading happens on a worker node?